framework_laptop.c

```diff
@@ -335,6 +687,26 @@ static int framework_probe(struct platform_device *pdev)
 	}
 #endif
 
+	struct cros_ec_device *ec = dev_get_drvdata(ec_device);
+	if (ec->cmd_readmem) {
+
+        // Count the number of fans
+        size_t fan_count;
+        if (ec_count_fans(&fan_count) < 0) {
+            printk(KERN_WARNING DRV_NAME ": failed to count fans.\n");
+            return -EINVAL;
+        }
+        // NULL terminates the list after the last detected fan
+        fw_hwmon_attrs[fan_count * FW_ATTRS_PER_FAN] = NULL;
+
+        hwmon_dev = hwmon_device_register_with_groups(dev, DRV_NAME, NULL, fw_hwmon_groups);
+        if (IS_ERR(hwmon_dev))
+            return PTR_ERR(hwmon_dev);
+
+    } else {
+		dev_err(dev, DRV_NAME ": Fan readings could not be enabled for this EC %s.\n", FRAMEWORK_LAPTOP_EC_DEVICE_NAME);
```

Comment:
nit: lowercase `Fan` for consistency with other messages

```suggestion
		dev_err(dev, DRV_NAME ": fan readings could not be enabled for this EC %s.\n", FRAMEWORK_LAPTOP_EC_DEVICE_NAME);
```